### ISSUE TITLE
Map shaking after setMaxBounds and setMinZoom

### ISSUE THREAD

**Issue by SashaLifashkin** [Leaflet/Leaflet#8532]
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

1. Create a map.
2. Set its maximum borders and minimum zoom to those currently visible on the screen.

`const map = L.map('map').setView([50.450036, 30.5241361], 13)`

`L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {`
`    attribution:`
`        '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',`
`}).addTo(map)`

`map.setMaxBounds(map.getBounds()).setMinZoom(map.getZoom())`

### Expected behavior

The map is limited by the maximum borders and the minimum zoom without any unexpected behavior.

### Current behavior

Map is limited by the maximum borders and the minimum zoom, but the map is moving (shaking).
I have looked at these issues: #[1946](https://github.com/Leaflet/Leaflet/pull/1946), #[2107](https://github.com/Leaflet/Leaflet/issues/2107), #[2011](https://github.com/Leaflet/Leaflet/issues/2011), #[2081](https://github.com/Leaflet/Leaflet/issues/2081), #[1866](https://github.com/Leaflet/Leaflet/issues/1866), #[2187](https://github.com/Leaflet/Leaflet/pull/2187). But it seems that there is no solution to my problem there, so I don't think it's a duplicate to any of previous issues. If it is, could you please help me to resolve this issue?
Thank you!

### Minimal example reproducing the issue

https://jsfiddle.net/fd3mertj/1/

### Environment

- Leaflet version:  1.8.0, 1.9.1
- Browser (with version): Mozilla 105.0.1 (64 bit), Opera 90.0.4480.84
- OS/Platform (with version): Windows 10 laptop

**Comment by Malvoz** [comment]
Can reproduce in Chrome 106.0.5249.91 on Windows 10, I see a very small but noticeable wobble, stops for me if window is resized. Also sometimes had to resize the window to reproduce.

**Comment by druwan** [comment]
I believe that the issue comes from the dynamic restriction of `setMaxBounds`. So a quick workaround to stop the shaking that worked for me, was to change the geographical bound of `setBounds`

Maybe it can work for you too until someone with more knowledge joins the discussion. 😄

**Comment by Falke-Design** [comment]
@SashaLifashkin can you please share a video

**Comment by SashaLifashkin** [comment]
> @SashaLifashkin can you please share a video

https://jsfiddle.net/fd3mertj/1/
From this link and some of the code I added, the problem is visible.

**Comment by SashaLifashkin** [comment]
> I believe that the issue comes from the dynamic restriction of `setMaxBounds`. So a quick workaround to stop the shaking that worked for me, was to change the geographical bound of `setBounds`
> 
> Maybe it can work for you too until someone with more knowledge joins the discussion. 😄

Unfortunately this solution didn't work for me.

**Comment by mourner** [comment]
@rjackson great analysis! Please submit a PR of course.

**Comment by jonkoops** [comment]
Do we perhaps feel like this fix deserves a backport for v1?

**Comment by Falke-Design** [comment]
Don't know, I think we should look that we get away from `v1`

**Comment by jonkoops** [comment]
Agreed, however #8536 was already cherry-picked, so I think we're going to do a release anyways.

**Comment by Falke-Design** [comment]
Yes that's true. Let us cherry pick this, because we make a release anyway

**Comment by jonkoops** [comment]
Done 👍
